core/src/main/scala/scalaz/std/String.scala

```diff
@@ -48,9 +48,8 @@ trait StringFunctions {
 
   // Parsing functions.
   private def intChars(s: String): Boolean = (s.length > 0) && (s.forall(c => c.isDigit || c == '-'))
-  private def floatChars(s: String): Boolean = (s.length > 0) && (s.forall(c => c.isDigit || c == '-' || c == '.'))
 
-  private def asNumber[T](f: String => T, b: String => Boolean, r: Tuple2[T, T], s: String)(implicit t: ClassTag[T]): Validation[String, T] =
+  private def asNumber[T](f: String => T, r: Tuple2[T, T], s: String)(implicit t: ClassTag[T]): Validation[String, T] =
```

Comment:
the `b` parameter must not be removed. It need to be used in place of `intChars`. (so that `floatChars` can be use as an alternative, by `parseDouble` and `parseFloat`)